pandapower/plotting/patch_makers.py

```diff
@@ -257,29 +300,112 @@ def sgen_patches(node_coords, size, angles, **kwargs):
+
+        if len(unique_angles) != 0 and "PV" in unique_angles[i]["sgen"]:
+            try:
+                angle = unique_angles[i]["sgen"]["PV"] + (angles[i] or 0)
+            except IndexError:
+                continue
+            mid_rect = node_geo + _rotate_dim2(np.array([0, 2 * size]), angle)
+            rect_lbottom = [mid_rect[0] - (pv_rect_size / 4), mid_rect[1]]
+            pv_patch = Rectangle((rect_lbottom), pv_rect_size / 2, pv_rect_size, angle=-angle * (180 / np.pi),
+                                 rotation_point=(mid_rect[0], mid_rect[1]), ec="k", fc="none")
+            polys.append(pv_patch)
+            lines.append((node_geo, mid_rect))
+            triangle_base = pv_tri_size
+            triangle_height = pv_tri_size * 0.75
+            triangle_points = [
+                (mid_rect + _rotate_dim2(np.array([- triangle_base / 2, + 2 * size]), angle)),
+                (mid_rect + _rotate_dim2(np.array([+ triangle_base / 2, + 2 * size]), angle)),
+                (mid_rect + _rotate_dim2(np.array([0, + 1.5 * triangle_height]), angle)),
+            ]
+            triangle_patch = Polygon(triangle_points, ec="k", fc="none")
+            polys.append(triangle_patch)
+
+        if len(unique_angles) != 0 and "wye" in unique_angles[i]["sgen"]:
+            try:
+                angle = unique_angles[i]["sgen"]["wye"] + (angles[i] or 0)
+            except IndexError:
+                continue
+            mid_circ = node_geo + _rotate_dim2(np.array([0, offset + size]), angle)
+            circ_edge = node_geo + _rotate_dim2(np.array([0, offset]), angle)
+            mid_tri1 = mid_circ + _rotate_dim2(np.array([r_triangle, -r_triangle / 4]), angle)
+            mid_tri2 = mid_circ + _rotate_dim2(np.array([-r_triangle, r_triangle / 4]), angle)
+            # dropped perpendicular foot of triangle1
+            perp_foot1 = mid_tri1 + _rotate_dim2(np.array([0, -r_triangle / 2]), angle)
+            line_end1 = perp_foot1 + + _rotate_dim2(np.array([-2.5 * r_triangle, 0]), angle)
+            perp_foot2 = mid_tri2 + _rotate_dim2(np.array([0, r_triangle / 2]), angle)
+            line_end2 = perp_foot2 + + _rotate_dim2(np.array([2.5 * r_triangle, 0]), angle)
+            polys.append(Circle(mid_circ, size, fc=facecolors[i], ec=edgecolors[i]))
+            polys.append(RegularPolygon(mid_tri1, numVertices=3, radius=r_triangle,
+                                        orientation=-angle, fc=facecolors[i], ec=edgecolors[i]))
+            polys.append(RegularPolygon(mid_tri2, numVertices=3, radius=r_triangle,
+                                        orientation=np.pi - angle, fc=facecolors[i],
+                                        ec=edgecolors[i]))
+            lines.append((node_geo, circ_edge))
+            lines.append((perp_foot1, line_end1))
+            lines.append((perp_foot2, line_end2))
+        else:
+            continue
```

Comment:
Is there any option for plotting without "unique_angles"? To me it looks like there isn't. I also think that defining the type of patch you want to draw via a variable called "unique_angles" is not very transparent. I would prefer some sort of "type" lookup that defines the type of patch used for each sgen and its angle. If it is not given for any of the indices, that sgen has the default patch.